fmf.spec

```diff
@@ -20,22 +23,12 @@ with support for inheritance and elasticity it provides an
 efficient way to organize data into well-sized text documents.
 This package contains the command line tool.
 
-%?python_enable_dependency_generator
-
 
-%package -n     python%{python3_pkgversion}-%{name}
+%package -n     python3-fmf
```

Comment:
We can do the same as in tmt and have only fmf package (providing python-fmf)

Reply:
Oh, you mean like: https://github.com/teemtee/tmt/blob/b782a6067f4c2f8f2a10318edd3a6b06ccced5de/tmt.spec#L133

Sure, just commenting with the link here so I have some reference to do that :D

Reply:
Wait, why do we want to do it one way instead of the other? `fmf` behaves like a standard python package, so shouldn't it be plain `python3-fmf`. @martinhoyer @lukaszachy @psss which way should it go, and is it even possible to `Obsoletes: fmf` given that `fmf` is the main package (it would be possible if the dist-git package is renamed to `python-fmf`)

Reply:
You don't actually obsolete fmf or python-fmf, they will just become one package, if it make sense.  
There was a discussion about it in tmt PR I can dig-up if you want.   
Also, the bin/fmf is now part of fmf anyway here :)  

I see there is discussion from October - sorry for opening another one.  

FYI, I believe the `%py_provides python3-fmf` is not needed to be explicitly set. At least when I tested it in https://github.com/teemtee/tmt/pull/2880#, I was able to install the package with both `dnf install tmt` and `dnf install python3-tmt`.

Reply:
> You don't actually obsolete fmf or python-fmf

I think you need to obsoletes in order for the `dnf upgrade` to work well. I think this would be caught by `installability` tests, but we can't run them in upstream, but they should be available in a dist-git PR.

> I see there is discussion from October - sorry for opening another one.

My memory also barely goes back 1-2 months back :sweat_smile:

> FYI, I believe the %py_provides python3-fmf is not needed to be explicitly set.

Indeed, that should be handled by `python3-devel` iiuc, or just the presence of python package metadata. For reference: [cmake PR adding python metadata](https://src.fedoraproject.org/rpms/cmake/pull-request/40)

Reply:
> My memory also barely goes back 1-2 months back 😅

Months? I can barely do hours :D 

> I think you need to obsoletes in order for the dnf upgrade to work 

Maybe. I'd check the tmt PR, as that was quite comprehensively reviewed and tested.

Reply:
Split this to https://github.com/teemtee/fmf/pull/241since that is more release dependent and may require some more discussion.